specs/capella/beacon-chain.md

```diff
@@ -323,11 +328,23 @@ def process_withdrawals(state: BeaconState, payload: ExecutionPayload) -> None:
     for expected_withdrawal, withdrawal in zip(expected_withdrawals, payload.withdrawals):
         assert withdrawal == expected_withdrawal
         decrease_balance(state, withdrawal.validator_index, withdrawal.amount)
-    if len(expected_withdrawals) > 0:
+
+    # Update the next withdrawal index if this block contained withdrawals
+    if len(expected_withdrawals) != 0:
         latest_withdrawal = expected_withdrawals[-1]
         state.next_withdrawal_index = WithdrawalIndex(latest_withdrawal.index + 1)
+
+    # Update the next validator index to start the next withdrawal sweep
+    if len(expected_withdrawals) == MAX_WITHDRAWALS_PER_PAYLOAD:
```

Comment:
For the `len(expected_withdrawals) == MAX_WITHDRAWALS_PER_PAYLOAD` case `state.next_withdrawal_index` is updated twice. Is that okay? Why not do
```
if len == MAX_WITHDRAWALS_PER_PAYLOAD
elif len > 0
else
```

Reply:
Where is the second update? I only see it set once.

Reply:
Oh, miss-read `next_withdrawal_index` for `next_withdrawal_validator_index`. All good then :+1:

Reply:
the names are confusing to read as is :/

Reply:
agreed :/